helper/tests/acc/provisioners.go

```diff
@@ -0,0 +1,130 @@
+package acc
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/packer/helper/tests"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	amazonEBS "github.com/hashicorp/packer/builder/amazon/ebs/acceptance"
+	virtualboxISO "github.com/hashicorp/packer/builder/virtualbox/iso/acceptance"
+	"github.com/hashicorp/packer/command"
+	"github.com/hashicorp/packer/packer"
+)
+
+func TestProvisionersAgainstBuilders(provisionerAcc ProvisionerAcceptance, t *testing.T) {
+	provisioner := provisionerAcc.GetName()
+	builders := checkBuilders(t)
+
+	// build template file and run a build for each builder with the provisioner
+	for _, builder := range builders {
+		builderAcc := BuildersAccTest[builder]
+		builderConfigs, err := builderAcc.GetConfigs()
+		if err != nil {
+			t.Fatalf("bad: failed to read builder config: %s", err.Error())
+		}
+
+		for vmOS, builderConfig := range builderConfigs {
+			if !provisionerAcc.IsCompatible(builder, vmOS) {
+				continue
+			}
+
+			testName := fmt.Sprintf("testing %s builder against %s provisioner", builder, provisioner)
+			t.Run(testName, func(t *testing.T) {
+				provisionerConfig, err := provisionerAcc.GetConfig()
```

Comment:
Would it be a possibility/helpful here to make this test parallel ?
By adding a `t.Parallel()` line; I'm suggesting this because on the pr description you mentionned:

> For some reason amazon-ebs can't run first, it doesn't seem to finish properly and causes later builders problems when decoding the template.

And so by making the tests run in Parallel then a failing test wont impact other test as it is run in another goroutine?

Or could this may be due to sort of a race condition due to the fact that some core packer object is being reused ?

There's two option:
```go
			testName := fmt.Sprintf("testing %s builder against %s provisioner", builder, provisioner)
			t.Parallel() // here
			t.Run(testName, func(t *testing.T) {
				t.Parallel() // or here
				provisionerConfig, err := provisionerAcc.GetConfig()
```

Reply:
A third option would be to make sure each builder is tested in a different circle-ci job

Reply:
I think is a race condition because the test doesn't fail. The other builder uses some core config from the previous one and fails because it can't identify the configs. Switching the run order it works 👀 . 
I don't know why I thought the t.Run was parallel 🤔  I can test with t.Parallel and see what happens. Should we define the max number of go routines as well?

Reply:
Well `t.Run` does start a goroutine but then it's limited to  one `Run` at a time if `t.Parallel` isn't called. :D 

Ah but if it is a race condition then I think that calling `Parallel` will only make things worse. Because each builder/provisioner is meant to be a single use exec.

Reply:
The less tricky option would be to make sure each one of those are run from a different process 🤔